biogeochem/EDPhysiologyMod.F90

```diff
@@ -281,8 +282,17 @@ subroutine trim_canopy( currentSite )
                          (EDPftvarcon_inst%grperc(ipft) + 1._r8)
                 else !evergreen costs
                    ! Leaf cost at leaf level z accounting for sla profile
-                   currentCohort%leaf_cost = 1.0_r8/(sla_levleaf* &
+		   if(use_leaf_age==itrue)then
+                      currentCohort%leaf_cost = 1.0_r8/(sla_levleaf* &
+                        (currentCohort%fracExpLeaves*EDPftvarcon_inst%expleaf_flong(ipft)+ &
+			 currentCohort%fracYoungLeaves*EDPftvarcon_inst%youngleaf_flong(ipft) + &
+			 currentCohort%fracOldLeaves*EDPftvarcon_inst%oldleaf_flong(ipft) + &
+			 currentCohort%fracSenLeaves*EDPftvarcon_inst%senleaf_flong(ipft) ) & 
+			 *EDPftvarcon_inst%leaf_long(ipft)*1000.0_r8) !convert from sla in m2g-1 to m2kg-1		   
```

Comment:
@xuchongang could this be written in an extensible way with 'n' pools of leave and the frac_leaf_age as a vector?

Reply:
@rgknox , I rember you told me that there will be error if you put arrays on the cohort. If it is Ok, I can easily update the codes to do that.

Reply:
shouldn't this leaf cost term be the same logic as the maintenance demand calculated as in https://github.com/NGEET/fates/pull/432/files#diff-b2580903c042bbb143bb267a7b882e3fR1120; i.e. based on the senescent pool and turnover rate?

Reply:
Also, isn't this rate supposed to represent the average, or time-integrated, turnover rate?  We are using the instantaneous fractions of leaf-ages for this calculation that is performed once per year.  Although, I suppose that if a plant is keeping up with its allocation and is continually on allometry, then this is a steady state problem?

Reply:
@rgknox agreed and keep the turnover based on the mean leaf longevity will resolve this issue.